src/models/room-state.js

```diff
@@ -197,6 +197,21 @@ RoomState.prototype.setStateEvents = function(stateEvents) {
                 member.setPowerLevelEvent(event);
                 self.emit("RoomState.members", event, self, member);
             });
+
+            // Go through the sentinel members and see if any of them would be
+            // affected by the new power levels. If so, replace the sentinel.
+            for (const userId of Object.keys(self._sentinels)) {
+                const oldSentinel = self._sentinels[userId];
+                const newSentinel = new RoomMember(event.getRoomId(), userId);
+                newSentinel.setMembershipEvent(oldSentinel.events.member, self);
```

Comment:
can we copy the old sentinel in a way that doesn't result in `RoomMember.powerLevel`s being raised for every member of the room?

Reply:
Since we don't emit `RoomState.newMember` here, the event won't be re-emitted by the client, won't go anywhere as no listeners have been added. Seems like this is probably the right semantics.

Reply:
oh, I see. I suppose. Comment might help.